### DISCUSSION TITLE
Conversion to plain Rust types

### DISCUSSION THREAD

**Discussion by giacomocariello** [tweag/nickel#786] [Q&A]
I've approached to nickel-lang from Dhall. Dhall's support for Rust includes serde_dhall crate, which allows to convert a Dhall configuration to rust plain types which implement serde Deserialize trait. I was wondering if there's a similar feature in nickel-lang or if it's planned.

**Comment by yannham** [comment]
Hello,

It doesn't currently exists, but should be relatively easy to do, because Nickel itself uses serde under the hood to convert an evaluated expression to JSON, YAML, etc. So the effort would be mostly plumbing things together. It's currently not on the short term roadmap, though. I can also happily provide help to any contributors willing to implement such a library.

**Reply by giacomocariello** [reply]
Hi @yannham, I've made a first attempt at implementing the aforementioned feature and posted a PR: https://github.com/tweag/nickel/pull/790.
